### PR TITLE
fix: 2 bugs manifested in walking content from policy-overrides: atte…

### PR DESCRIPTION
**Pull request by naugtur**
…mpt to go down `$root$` and failure to load content from the main of the package

### CODE REVIEW

File: packages/core/src/generatePolicy.js [hunk 1]
```diff
@@ -74,6 +74,9 @@ function createModuleInspector(opts) {
     moduleRecord,
     { isBuiltin, includeDebugInfo = false }
   ) {
+    if (moduleRecord === undefined) {
```
Comment (naugtur) [inline packages/core/src/generatePolicy.js:77]: This is redundant for the bugfix in question but it's the correct thing to do here anyway.

File: packages/lavamoat-node/src/parseForPolicy.js [hunk 2]
```diff
@@ -186,14 +186,22 @@ function makeImportHook({
...
     const packageName = getPackageNameForModulePath(canonicalNameMap, filename)
 
-    if (commonjsExtensions.includes(extension)) {
-      return makeJsModuleRecord(specifier, filename, packageName)
-    }
     if (extension === '.node') {
       return makeNativeModuleRecord(specifier, filename, packageName)
     }
+    try {
+      var content = await fs.readFile(filename, 'utf8')
```
Comment (naugtur) [inline packages/lavamoat-node/src/parseForPolicy.js:193]: intentional use of var for nicer code.

File: packages/lavamoat-node/src/parseForPolicy.js [hunk 3]
```diff
@@ -264,8 +270,18 @@ function makeImportHook({
...
         }
+        // do not pursue root as a dependency to scan
+        if (packageName === '$root$') {
+          return
+        }
         // resolve and add package main in override
         const packageRoot = getMapKeyForValue(canonicalNameMap, packageName)
+        if (!packageRoot) {
+          console.warn(
+            `could not find package root for ${packageName} as found in policy-override`
```
Comment (naugtur) [inline packages/lavamoat-node/src/parseForPolicy.js]:
```suggestion
            `lavamoat could not find package's entry script for ${packageName} as found in policy-override`
```

**Review by weizman — APPROVED** [review]
Lgtm

**Pull request merged**